File: hw/tipantiza/u3/hw18OOPConceptsZoo/Java/OOPConcepts/src/main/java/ec/edu/espe/oopconcepts/utils/MongoConnection.java
package ec.edu.espe.oopconcepts.utils;

import com.mongodb.client.*;
import org.bson.Document;

/**
 *
 * @author Alexander Tipantiza, The Softwarrios, @ESPE
 */
public class MongoConnection {

    private static String URI = "mongodb+srv://tipantizaalexander:Alexander20@cluster0.z86uqo3.mongodb.net/?appName=Cluster0";
    private static String DATABASE_NAME = "FarmChickenSystem";

    private static MongoDatabase database;

    public static MongoDatabase getDatabase() {

        try {

            if (database == null) {

                MongoClient client = MongoClients.create(URI);
                database = client.getDatabase(DATABASE_NAME);

                System.out.println("Connected to MongoDB");

            }

        } catch (Exception e) {
            System.out.println("Error connecting to MongoDB: " + e.getMessage());
        }

        return database;
    }
}

File: hw/tipantiza/u3/hw18OOPConceptsZoo/Python/OOPConcepts/utils/mongo_connection.py
from pymongo import MongoClient


class MongoConnection:

    URI = "mongodb+srv://tipantizaalexander:Alexander20@cluster0.z86uqo3.mongodb.net/?appName=Cluster0"
    DATABASE_NAME = "FarmSistemPy"

    _database = None

    @staticmethod
    def get_database():

        try:

            if MongoConnection._database is None:

                client = MongoClient(MongoConnection.URI)

                MongoConnection._database = client[MongoConnection.DATABASE_NAME]

                print("Connected to MongoDB")

        except Exception as e:

            print(f"Error connecting to MongoDB: {e}")

        return MongoConnection._database